scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/kms/frontend/App.scala

```diff
@@ -0,0 +1,290 @@
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.ppml.kms.frontend
+
+import java.io.File
+import java.security.{KeyStore, SecureRandom}
+import java.util.concurrent.TimeUnit
+import javax.net.ssl.{KeyManagerFactory, SSLContext, TrustManagerFactory}
+
+import akka.actor.{ActorRef, ActorSystem, Props}
+import akka.http.scaladsl.{ConnectionContext, Http}
+import akka.http.scaladsl.server.Directives.{complete, path, _}
+import akka.pattern.ask
+import akka.stream.ActorMaterializer
+import akka.util.Timeout
+import com.codahale.metrics.{MetricRegistry, Timer}
+import com.intel.analytics.bigdl.orca.inference.EncryptSupportive
+import com.intel.analytics.bigdl.serving.utils.Conventions
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable
+import scala.concurrent.Await
+import com.intel.analytics.bigdl.dllib.utils.Log4Error
+import sys.process._
+import java.util.Base64
+import javax.crypto.spec.SecretKeySpec
+import javax.crypto.Cipher
+
+object App extends Supportive {
+  override val logger = LoggerFactory.getLogger(getClass)
+
+  val name = "BigDL KMS Frontend"
+
+  implicit val system = ActorSystem("bigdl-kms-frontend-system")
+  implicit val materializer = ActorMaterializer()
+  implicit val executionContext = system.dispatcher
+  implicit val timeout: Timeout = Timeout(100, TimeUnit.SECONDS)
+
+  def main(args: Array[String]): Unit = {
+      val arguments = timing("parse arguments")() {
+        argumentsParser.parse(args, KMSFrontendArguments()) match {
+          case Some(arguments) => logger.info(s"starting with $arguments"); arguments
+          case None => argumentsParser.failure("miss args, please see the usage info"); null
+        }
+      }
+      logger.info("Servable Manager Load Success!")
+      val route = timing("initialize http route")() {
+        path("") {
+          timing("welcome")(overallRequestTimer) {
+            complete("welcome to " + name)
+          }
+        } ~ (get & path("generatePrimaryKey") &
+          extract(_.request.entity.contentType) & entity(as[String])) {
+          (contentType, content) => {
+           timing("generatePrimaryKey")(overallRequestTimer) {
+            try{
+                val params = content.split("&")
+                val primaryKeyName = params(0).split("=")(1)
+                val user = params(1).split("=")(1)
+                val password = params(2).split("=")(1)
+                val base64AES256Key:String = generateAESKey(256)
+                login(user, password)
+                addKeyToKeywhiz(user, primaryKeyName, base64AES256Key)
+                complete(s"primaryKey [$primaryKeyName] is generated successfully!")
+            } catch {
+              case e: Exception =>
+                e.printStackTrace()
+                complete(500, e.getMessage + "\n please get a primary key like: " +
+                  "primaryKeyName=a_primary_key_name&user=your_username&password=your_password")
```

Comment:
does the order of keys matter?
please refer to
https://doc.akka.io/docs/akka-http/current/routing-dsl/directives/parameter-directives/parameters.html